sdk/src/Services/RDS/Custom/Util/RDSAuthTokenGenerator.cs

```diff
@@ -85,7 +86,7 @@ public static string GenerateAuthToken(string hostname, int port, string dbUser)
         /// <returns></returns>
         public static string GenerateAuthToken(RegionEndpoint region, string hostname, int port, string dbUser)
         {
-            AWSCredentials credentials = FallbackCredentialsFactory.GetCredentials();
+            AWSCredentials credentials = DefaultIdentityResolverConfiguration.ResolveDefaultIdentity<AWSCredentials>();
```

Comment:
Could this be `DefaultAWSCredentialsIdentityResolver` instead? Then customers could call get credentials directly (and I assume most people wouldn't know about the `AWSCredentials` abstract class).

(This comment isn't about this specific helper, but thinking about how we'd tell customers to stop using the `FallbackCredentialsFactory` after it's deprecated).

Reply:
I think replacing this call with `DefaultAWSCredentialsIdentityResolver.ResolveAWSCredentials()` makes sense. I initially thought we might need to resolve other types of identities, but so far I've only used it to resolve `AWSCredentials`.

Reply:
Talked offline, but OK not to change this for now (although for the migration guide we should check if this is the best approach, there are customers using the `FallbackCredentialsFactory` directly: https://github.com/search?q=FallbackCredentialsFactory.GetCredentials+NOT+repo%3Aaws%2Faws-sdk-net&type=code)